node/src/components/consensus/era_supervisor.rs

```diff
@@ -472,43 +474,38 @@ where
     pub(super) fn handle_validators_response(
         &mut self,
         block_header: BlockHeader,
-        validators: Option<ValidatorWeights>,
+        validator_weights: ValidatorWeights,
     ) -> Effects<Event<I>> {
-        match validators {
-            Some(validator_weights) => {
-                let validator_stakes = validator_weights
-                    .into_iter()
-                    .map(|(key, stake)| {
-                        (
-                            key.try_into().ok().expect("keys should convert correctly"),
-                            Motes::new(stake),
-                        )
-                    })
-                    .collect();
-                self.era_supervisor
-                    .current_era_mut()
-                    .consensus
-                    .deactivate_validator();
-                let new_era_id = block_header.era_id().successor();
-                info!(?new_era_id, "Era created");
-                let results = self.era_supervisor.new_era(
-                    new_era_id,
-                    Timestamp::now(), // TODO: This should be passed in.
-                    validator_stakes,
-                    block_header.timestamp(),
-                    block_header.height() + 1,
-                    *block_header.global_state_hash(),
-                );
-                let mut effects = self.handle_consensus_results(new_era_id, results);
-                effects.extend(
-                    self.effect_builder
-                        .announce_block_handled(block_header)
-                        .ignore(),
-                );
-                effects
-            }
-            None => todo!(),
-        }
+        let validator_stakes = validator_weights
+            .into_iter()
+            .map(|(key, stake)| {
+                (
+                    key.try_into().ok().expect("keys should convert correctly"),
```

Comment:
As discussed, this should not panic if it's possible to bond to an invalid key.

Reply:
Hmm... What do you think we should do here? `filter_map`, maybe, and log a warning/error? Or something "stronger"?

Reply:
I think filtering and logging would be fine for now.
I'd _like_ to slash the invalid validator… :smiling_imp: …but of course if this actually happens, it's most likely by mistake. I think there was also a plan to add some validation step to the bonding process, where the key would have to sign something before it was accepted?

Reply:
I'm not sure if there was such a plan, but it certainly sounds reasonable! I'll go with filtering and warning for now.